core/templates/pages/learner-dashboard-page/new-home-tab.component.ts

```diff
@@ -0,0 +1,263 @@
+   templateUrl: './new-home-tab.component.html',
+   styleUrls: ['./new-home-tab.component.css']
+ })
+export class NewHomeTabComponent {
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() currentGoals!: LearnerTopicSummary[];
+  @Input() goalTopics!: LearnerTopicSummary[];
+  @Input() partiallyLearntTopicsList!: LearnerTopicSummary[];
+  @Input() untrackedTopics!: Record<string, LearnerTopicSummary[]>;
+  @Input() username!: string;
+  @Input() explorationPlaylist!: LearnerExplorationSummary[];
+  @Input() incompleteExplorationsList!: LearnerExplorationSummary[];
+  @Input() completedExplorationsList!: LearnerExplorationSummary[];
+  @Input() collectionPlaylist!: CollectionSummary[];
+  @Input() storyIdToLearnerGroupsTitleMap!: Map<string, string[]>;
+  @Input() learnerGroupFeatureIsEnable: boolean = false;
+
+  carouselClassname: string = 'home-tab';
+  currentGoalsLength!: number;
+  classroomUrlFragment!: string;
+  goalTopicsLength!: number;
+  noPlaylistActivity: boolean = false;
+  showMoreInPlaylistSection: boolean = false;
+  CLASSROOM_LINK_URL_TEMPLATE: string = '/learn/<classroom_url_fragment>';
+  nextIncompleteNodeTitles: string[] = [];
+  widthConst: number = 233;
+  continueWhereYouLeftOffList: LearnerTopicSummary[] = [];
+  windowIsNarrow: boolean = false;
+  storyInProgress: storySummaryTile[] = [];
+  storyInRecommended: storySummaryTile[] = [];
+  expIds: string[] = [];
+  completedExpIds: string[] = [];
+  explorationToProgressMap: Map<string, number> = new Map();
+  totalExploration: LearnerExplorationSummary[] = [];
+  storyExplorationProgressList: (
+    LearnerExplorationSummary | storySummaryTile)[] = [];
+
+
+  totalLessonsInPlaylist: (
+      LearnerExplorationSummary | CollectionSummary)[] = [];
+
+  directiveSubscriptions = new Subscription();
+
+  constructor(
+    private i18nLanguageCodeService: I18nLanguageCodeService,
+    private learnerDashboardActivityBackendApiService:
+    LearnerDashboardActivityBackendApiService,
+    private windowDimensionService: WindowDimensionsService,
+    private urlInterpolationService: UrlInterpolationService,
+    private readOnlyExplorationBackendApiService:
+    ReadOnlyExplorationBackendApiService,
+  ) {}
+
+  ngOnInit(): void {
+    var allGoals = [...this.currentGoals, ...this.partiallyLearntTopicsList];
+    this.currentGoalsLength = this.currentGoals.length;
+    this.goalTopicsLength = this.goalTopics.length;
+
+    if (allGoals.length !== 0) {
+      var allGoalIds = [];
+      for (var goal of allGoals) {
+        allGoalIds.push(goal.id);
+      }
+      var uniqueGoalIds = Array.from(new Set(allGoalIds));
+      for (var uniqueGoalId of uniqueGoalIds) {
+        var index = allGoalIds.indexOf(uniqueGoalId);
+        this.continueWhereYouLeftOffList.push(allGoals[index]);
+      }
+    }
```

Comment:
why let is not be used in the place of var? (as per as the previous review)

Reply:
Done